File: automatos/csharp/Automata.Tests/Automata/AutomataFactory.cs
using System;
using Automata;

namespace Automata
{

	public static class AutomataFactory
	{
		/// <summary>
		/// Cria um autômato não-determinístico.
		/// </summary>
		/// <returns>The automata.</returns>
		public static INFA CreateNFA ()
		{
			return new NFA ();
		}

		/// <summary>
		/// 
		/// </summary>
		/// <returns>The DF.</returns>
		public static IDFA CreateDFA ()
		{
			return new DFA ();
		}
	}

}

File: automatos/java/automata/src/main/java/br/com/pucgo/automata/AutomataFactory.java
package br.com.pucgo.automata;

/**
 * A fábrica de autômatos.
 *
 * Created by danfma on 22/11/14.
 */
public final class AutomataFactory {

    /**
     * Cria um novo autômato NFA.
     *
     * @return a instância do autômato.
     */
    public static Automata createNonDeterministicAutomata() {
        return new NFA(); // substitua pela sua implementação
    }

}
